File: benchmarks/reference/python/sieve.py
import time

def sieve(n):
    flags = [True] * n
    count = 0
    p = 2
    while p < n:
        if flags[p]:
            count += 1
            m = p + p
            while m < n:
                flags[m] = False
                m += p
        p += 1
    return count

n = 500000
start = time.monotonic_ns()
result = sieve(n)
elapsed = time.monotonic_ns() - start
ms = elapsed // 1000000
print(f"primes below {n}: {result}")
print(f"elapsed: {ms} ms")

File: benchmarks/reference/go/sieve.go
package main

import (
	"fmt"
	"time"
)

func sieve(n int64) int64 {
	flags := make([]bool, n)
	for i := range flags {
		flags[i] = true
	}

	var count int64
	var p int64
	for p = 2; p < n; p++ {
		if flags[p] {
			count++
			for m := p + p; m < n; m += p {
				flags[m] = false
			}
		}
	}
	return count
}

func main() {
	var n int64 = 500000
	start := time.Now()
	result := sieve(n)
	elapsed := time.Since(start).Milliseconds()
	fmt.Printf("primes below %d: %d\n", n, result)
	fmt.Printf("elapsed: %d ms\n", elapsed)
}
